Documentation/rust/quick-start.rst

```diff
@@ -23,13 +23,13 @@ at the time of writing, they are likely to not be recent enough.
 rustc
 *****
 
-A recent *nightly* Rust toolchain (with, at least, ``rustc``) is required,
-e.g. ``nightly-2021-05-29``. Our goal is to use a stable toolchain as soon
-as possible, but for the moment we depend on a handful of nightly features.
+A particular version (`1.54.0-beta.1`) of the Rust compiler is required.
```

Comment:
> * Is it necessary for all files to carry SPDX identifers? Can we for example document somewhere that all alloc files are MIT or APACHE-2.0 so we can keep the files unchanged?

AFAIK, they are required (or worse, a bigger text with the licence like in some third-party files etc.).

In any case, it is better to avoid that fight now, i.e. we can ask lawyers later. It is not a big deal anyway, and can be mechanically ignored in e.g. a script.

> * I actually find `-Dmissing_docs` not very helpful for dev. Perhaps it's better to warn about missing docs in Makefile and only reject them in CI (like many other warnings).

It is true that people will complain sooner than later, so it is best to change it. My thinking here was that the warning was fairly easy/stable, and that most driver developers will not need many `pub` definitions. I think I can ask -next to run the CI for most drivers, so it should be fine long-term even if it is a warning.

> * You also want to add `no_global_oom_handling` to `Rc::new` and `Arc::new` as they use non-fallible `box` syntax. You cannot actually use them because they'll trigger error when monomorphized, but it's better to just remove them.

Indeed.

Reply:
Hmm... Somehow I replied in the wrong place. Let me move the comment outside.